File: 739_Daily_Temperatures.cs
public class Solution {
    public int[] DailyTemperatures(int[] T) {
        Stack<int> s = new Stack<int>();
        int[] res = new int[T.Count()];
        for (int i = T.Count() - 1; i >= 0; --i) {
            while (s.Count() > 0 && T[s.Peek()] <= T[i])
                s.Pop();
            res[i] = s.Count() == 0 ? 0 : s.Peek() - i;
            s.Push(i);
        }
        return res;
    }
}

File: 739_Daily_Temperatures.py
class Solution:
    def dailyTemperatures(self, T: List[int]) -> List[int]:
        stack, res = [], [0] * len(T)
        for i in range(len(T) - 1, -1, -1):
            while stack and T[stack[-1]] <= T[i]:
                stack.pop()
            res[i] = 0 if len(stack) == 0 else stack[-1] - i
            stack.append(i)
        return res
